src/Arcus.Security.Providers.DockerSecrets/DockerSecretsSecretProvider.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Arcus.Security.Core;
+using GuardNet;
+using Microsoft.Extensions.Configuration.KeyPerFile;
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.FileProviders;
+
+namespace Arcus.Security.Providers.DockerSecrets
+{
+    /// <summary>
+    /// Represents an <see cref="ISecretProvider" /> that provides access to the Docker secrets mounted into the Docker container as files.
+    /// </summary>
+    public class DockerSecretsSecretProvider : ISecretProvider
+    {
+        private readonly KeyPerFileConfigurationProvider _provider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DockerSecretsSecretProvider"/> class.
+        /// </summary>
+        /// <param name="secretsDirectoryPath">The path inside the docker container where the secrets are located.</param>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="secretsDirectoryPath"/> is <c>null</c>or an empty string.</exception>
```

Comment:
```suggestion
        /// <exception cref="ArgumentException">Thrown when the <paramref name="secretsDirectoryPath"/> is blank.</exception>
```

We also throw on blank `string`s 😉 .